add brightness control

arthurcgusmao opened the pull request:
Added the option to adjust the brightness also for those who may be interested.

arthurcgusmao committed:
add brightness control

arthurcgusmao committed:
override redshift default

Redshift's default configuration (when using a config file) usually
prevents changes from being applied from the command line. Hence,
in this commit we kill the redshift process to make changed from
the command line.

Also, we get the current redshift's configuration as basis for the
changes that redshift-adjust will make.

maandree commented on src/redshift-adjust:
```diff
@@ -42,10 +42,14 @@ EOF
 
 if [ ! -e "${file_temp}" ]; then
-    echo 6500 > "${file_temp}"
+    temperature=$(redshift -p | grep -Pzo "(?<=Color temperature: ).*")
```
`-P`, `-z`, and `-o` are not in grep(1posix), can you use `sed` instead?

arthurcgusmao replied:
I am trying, but I cannot find how to get only the content of where the line is using only `sed`. Would
```shell-script
temperature=$(redshift -p | grep 'Brightness:' | sed -e 's/Brightness: \(.*\)/\1/')
```
suffice? If not, any idea?

maandree replied:
`temperature=$(redshift -p | sed -n 's/Brightness: \(.*\)/\1/p')`

maandree commented on src/redshift-adjust:
```diff
@@ -42,10 +42,14 @@ EOF
 if [ ! -e "${file_temp}" ]; then
-    echo 6500 > "${file_temp}"
+    temperature=$(redshift -p | grep -Pzo "(?<=Color temperature: ).*")
+    temperature=${temperature%?} # remove last char (K)
```
This should be combined with `sed` above, and the `K` should be explicit.

arthurcgusmao replied:
Here, do you mean that I should remove the K explicitly by using, e.g.,
```shell-script
redshift -p | grep 'Color temperature:' | sed -e 's/Color temperature: \(.*\)K/\1/'
```
or that I should leave the K in the file?

maandree commented on src/redshift-adjust:
```diff
@@ -42,10 +42,14 @@ EOF
 fi
 if [ ! -e "${file_bright}" ]; then
-    echo 1.0 > "${file_bright}"
+    brightness=$(redshift -p | grep -Pzo "(?<=Brightness: ).*")
```
As above.

maandree commented on src/redshift-adjust:
```diff
@@ -42,10 +42,14 @@ EOF
-    echo 1.0 > "${file_bright}"
+    brightness=$(redshift -p | grep -Pzo "(?<=Brightness: ).*")
+    echo $brightness > "${file_bright}"
+    $(pkill redshift) # killing redshift prevents default config to override new config
```
This should just be `pkill '^redshift$'`.

arthurcgusmao replied:
Thanks, your suggestion solved a bug I was experiencing that the screen would blink and the settings would not be applied in the first time I was running redshift-adjust and redshift was running in the background.

arthurcgusmao committed:
Fix: kill redshift process

Killing the redshift process using the new version does not incur in the
screen blinking and the redshift-adjust new temperature not being
applied. Thanks to @maandree, as suggested in https://github.com/maandree/redshift-adjust/pull/2#discussion_r342743168

arthurcgusmao committed:
`sed` instead of `grep` for better compatibility

As pointed out by @maandree in
https://github.com/maandree/redshift-adjust/pull/2#discussion_r360731105
, the grep options being used were not in grep(1posix), so switching to
sed allows for greater compatibility.

arthurcgusmao commented:
Thank you for your support @maandree. I have made all modifications by [this commit](https://github.com/arthurcgusmao/redshift-adjust/commit/a2a9f04458852ed459bd33682025c86a58a19efa) and I'll open a new pull request on that one.
